src/components/ShareMyLocationButton.js

```diff
@@ -28,20 +28,33 @@ const ShareMyLocationButton = ({ disabled = false }) => {
 
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      if (trackingId === null) {
-        const newTrackingId = navigator.geolocation.watchPosition(
-          handleSuccess,
-          handleError,
-        );
-        setTrackingId(newTrackingId);
-      }
+      const trackingId = navigator.geolocation.watchPosition(
+        handleSuccess,
+        handleError,
+      );
+
+      setTrackingId(trackingId);
+      setUserLocationShared(true);
```

Comment:
I believe this line `setUserLocationShared(true);`  should be inside `handleSuccess` because there is a scenario where you don't share your location (browser prompt) this state changes to `true` but you don't want to share the user location.